client/engine/engine.go

```diff
@@ -521,23 +521,23 @@ func (e *Engine) handlePaymentRequest(request PaymentRequest) (EngineEvent, erro
 	ee.PaymentChannelUpdates = append(ee.PaymentChannelUpdates, info)
 
 	se := protocols.SideEffects{MessagesToSend: protocols.CreateVoucherMessage(voucher, payee)}
-	return ee, e.executeSideEffects(se)
+	return ee, e.ExecuteSideEffects(se)
 }
 
 // sendMessages sends out the messages and records the metrics.
 func (e *Engine) sendMessages(msgs []protocols.Message) {
 	defer e.metrics.RecordFunctionDuration()()
 
 	for _, message := range msgs {
-		message.From = *e.store.GetAddress()
+		message.From = *e.store.GetAddress() // read from disk on each msg: bottleneck?
 		e.logMessage(message, Outgoing)
 		e.recordMessageMetrics(message)
 		e.msg.Send(message)
 	}
 }
 
-// executeSideEffects executes the SideEffects declared by cranking an Objective or handling a payment request.
-func (e *Engine) executeSideEffects(sideEffects protocols.SideEffects) error {
+// ExecuteSideEffects executes the SideEffects declared by cranking an Objective or handling a payment request.
+func (e *Engine) ExecuteSideEffects(sideEffects protocols.SideEffects) error {
```

Comment:
This change is suspect. Before this change, all `sideEffect`s were manufactured locally by the running engine. They were "trusted" data. Executing them is presumed-safe (bugs notwithstanding), and `executeSideEffects` has not needed to be explicitly defensive in its implementation.

Reply:
This was replaced with a more sensible PushMessage method, so incoming data is properly "scrutinized".